crescent/internal/handle_resp.py

```diff
@@ -46,20 +54,40 @@ async def handle_resp(event: InteractionCreateEvent):
             name = unwrap(option.options)[0].name
             options = unwrap(option.options)[0].options
 
-    command = bot._command_handler.registry[
+    command = _get_command(bot, name, interaction.guild_id, group, sub_group)
+    ctx = Context._from_command_interaction(interaction)
+    callback_params = _options_to_kwargs(interaction, options)
+
+    await command(ctx, **callback_params)
+
+
+def _get_command(
+    bot: Bot,
+    name: str,
+    guild_id: Optional[Snowflake],
+    group: Optional[str],
+    sub_group: Optional[str],
+) -> CommandCallback:
+    with suppress(KeyError):
+        return bot._command_handler.registry[
+            Unique(
+                name=name,
+                type=AppCommandType.CHAT_INPUT,
+                guild_id=guild_id,
+                group=group,
+                sub_group=sub_group,
+            )
+        ].callback
+
+    return bot._command_handler.registry[
         Unique(
             name=name,
             type=AppCommandType.CHAT_INPUT,
-            guild_id=interaction.guild_id,
+            guild_id=UNDEFINED,
             group=group,
             sub_group=sub_group,
         )
```

Comment:
This feels like a lot of duplication. Why would the KeyError be raised? It looks like the assumption is the key error occurs because guild_id is None/non-existent, so it might be best to check that explicitly. The current method is prone to bugs from another of the keys being invalid.

Reply:
The interaction always sends a guild id so i'm checking with and without a guild_id. I think it may be cleaner if i rework the system to use command IDs so I can try that.

Reply:
Okay. Something like this might also work:
```py
kwargs = dict(
    name=name,
    type=AppCommandType.CHAT_INPUT,
    group=group,
    sub_group=sub_group,
)

try:
    return bot._command_handler.registry[Unique(**kwargs, guild_id=guild_id)]
except KeyError:
    return bot._command_handler.register[Unique(**kwargs, guild_id=UNDEFINED)]
```

That way it's clear exactly what you're doing and why you're doing it.

Reply:
I ended up refactoring it to this
```python
    kwargs = dict(
        name=name,
        type=AppCommandType.CHAT_INPUT,
        group=group,
        sub_group=sub_group,
    )

    with suppress(KeyError):
        return bot._command_handler.registry[Unique(guild_id=guild_id, **kwargs)]  # type: ignore
    with suppress(KeyError):
        return bot._command_handler.registry[Unique(guild_id=UNDEFINED, **kwargs)]  # type: ignore
    raise CommandNotFoundError(f"Handler for command `{name}` does not exist locally.")
```

Reply:
What's the reason for the type ignores?

Reply:
mypy doesn't like that **kwargs are typed as object

Reply:
just changed it to Dict[str, Any] and now mypy likes it

Reply:
Just type the kwargs as `Dict[str, Any]`. That way you don't have type: ignore's